Only request WooPay session if we haven't already

bborman22 opened the pull request:
Fixes #7357 

#### Changes proposed in this Pull Request

<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

*

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

bborman22 committed:
Only request woopay session if we haven't already

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7358` or branch name `fix/7357-multiple-woopay-session-requests` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7357-multiple-woopay-session-requests) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 214a65a48a1b2a2060299297f85e0229d92b9427
- Build time: 2023-10-05 18:14:02 UTC

_Note: the build is updated when a new commit is pushed to this PR._

cesarcosta99 committed:
Revert "Only request woopay session if we haven't already"

This reverts commit e7c34cd27816cefdbdf96a0b0057afcf12c1ff96.

cesarcosta99 committed:
Prevent WooPay button from re-rendering when navigating away

The reason this occurred, was because an `addToCart()` HTTP request was made, which triggered the `updated_cart_totals` event to fire. This event then caused a new WooPay button to be rendered right before the browser redirected to the WooPay checkout page.

cesarcosta99 committed:
Add changelog entry

bborman22 committed:
Merge branch 'develop' into fix/7357-multiple-woopay-session-requests

cesarcosta99 committed:
Fix email selector

cesarcosta99 committed:
Change approach to use the new ReactDOM API and remove jQuery

cesarcosta99 committed:
Rename variable

cesarcosta99 committed:
Update import path

bborman22 commented:
Closing in favor of #7393
